control-plane/connect-inject/endpoints_controller.go

```diff
@@ -653,6 +665,96 @@ func (r *EndpointsController) createGatewayRegistrations(pod corev1.Pod, service
 	return serviceRegistration, nil
 }
 
+// createTerminatingGatewayRegistrations creates the gateway service registrations with the information from the Pod.
+func (r *EndpointsController) createTerminatingGatewayRegistrations(pod corev1.Pod, serviceEndpoints corev1.Endpoints, healthStatus string) (*api.CatalogRegistration, error) {
+	svcID := serviceID(pod, serviceEndpoints)
+
+	meta := map[string]string{
+		MetaKeyPodName:         pod.Name,
+		MetaKeyKubeServiceName: serviceEndpoints.Name,
+		MetaKeyKubeNS:          serviceEndpoints.Namespace,
+		MetaKeyManagedBy:       managedByValue,
+	}
+	for k, v := range pod.Annotations {
+		if strings.HasPrefix(k, annotationMeta) && strings.TrimPrefix(k, annotationMeta) != "" {
+			if v == "$POD_NAME" {
+				meta[strings.TrimPrefix(k, annotationMeta)] = pod.Name
+			} else {
+				meta[strings.TrimPrefix(k, annotationMeta)] = v
+			}
+		}
+	}
+
+	tags := consulTags(pod)
+
+	/*
+			 service {
+		✅	   kind = "terminating-gateway"
+			   name = "{{ .name }}"
+		✅	   id = "${POD_NAME}"
+			   {{- if $root.Values.global.enableConsulNamespaces }}
+			   namespace = "{{ (default $defaults.consulNamespace .consulNamespace) }}"
+			   {{- end }}
+			   {{- if $root.Values.global.adminPartitions.enabled }}
+			   partition = "{{ $root.Values.global.adminPartitions.name }}"
+			   {{- end }}
+		✅	   address = "${POD_IP}"
+		✅	   port = 8443
+			   {{- if (and $root.Values.global.metrics.enabled $root.Values.global.metrics.enableGatewayMetrics) }}
+			   proxy { config { envoy_prometheus_bind_addr = "${POD_IP}:20200" } }
+			   {{- end }}
```

Comment:
Need to thread through enableGatewayMetrics.
Get that value either at install or from an annotation.

Reply:
Get the enable gateway metrics into the metrics struct.